java-frontend/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -306,10 +354,23 @@ private void closeArguments(final Arguments arguments, int stackOffset) {
     }
 
     private void closeResource(@Nullable final SymbolicValue target) {
+      ObjectConstraint openConstraint = null;
       if (target != null) {
-        ObjectConstraint oConstraint = programState.getConstraintWithStatus(target, Status.OPENED);
-        if (oConstraint != null) {
-          programState = programState.addConstraint(target.wrappedValue(), oConstraint.withStatus(Status.CLOSED));
+        openConstraint = programState.getConstraintWithStatus(target, Status.OPENED);
+        if (openConstraint != null) {
+          programState = programState.addConstraint(target.wrappedValue(), openConstraint.withStatus(Status.CLOSED));
+        }
+      }
+      if (openConstraint == null) {
+        Optional<SymbolicValue> closeableParameter = closeableParameters.stream().filter(sv -> sv == target).findAny();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)